File: algorithm/leetcode5437.least-number-of-unique-integers-after-k-removals.cpp
class Solution {
public:
    int findLeastNumOfUniqueInts(vector<int>& arr, int k) {
    	unordered_map<int,int> m;
    	int i;
    	for(i = 0; i < arr.size(); ++i)
    		m[arr[i]]++;
    	vector<int> count(m.size());
    	i = 0;
    	for(auto it = m.begin(); it != m.end(); ++it)
    		count[i++] = it->second;
    	sort(count.begin(), count.end());
    	i = 0;
    	while(k > 0)
    	{
    		k -= count[i];
    		if(k >= 0)
    			i++;
    	}
    	return count.size()-i;
    }
};

File: algorithm/Python3_version/leetcode5437.least-number-of-unique-integers-after-k-removals.py
class Solution:
    def findLeastNumOfUniqueInts(self, arr: List[int], k: int) -> int:
        d = {}
        for i in arr:
            if i in d:
                d[i] += 1
            else:
                d[i] = 1
        count = list(d.values())
        # count.sort()
        count = sorted(count)
        i = 0
        for c in count:
            k -= c
            if k >= 0:
                i += 1
            if k <= 0:
                break
        return len(count) - i